<launch>
    <node name="orb_slam3_stereo" pkg="orb_slam3_ros" type="orb_slam3_ros_kitti_node" output="screen">

        <!-- for EuRoC dataset -->
        <remap from="/camera/left/image_raw"    to="/image_loader_zed_node/left"/>
        <remap from="/camera/right/image_raw"   to="/image_loader_zed_node/right"/>

        <param name="kitti_save_topic" type="string" value="/load_image/save"/>
        <param name="path" type="string"   value="$(find orb_slam3_ros)/traj/" />

        <param name="cone_detection" type="string" value="/cone_detection/output"/>
        <!-- Parameters for original ORB-SLAM3 -->
        <param name="voc_file"      type="string"   value="$(find orb_slam3_ros)/ORB_SLAM3/Vocabulary/ORBvoc.txt" />
        <param name="settings_file" type="string"   value="$(find orb_slam3_ros)/config/zed2i.yaml" />
    
        <!-- Parameters for ROS -->
        <param name="world_frame_id"    type="string"   value="map" />
        <param name="cam_frame_id"      type="string"   value="zed2i_left_camera_frame" />
        <param name="target_frame_id"   type="string"   value="zed2i_left_camera_frame"/>
        <param name="parent_frame_id"   type="string"   value="base_link"/>
        <param name="enable_pangolin" type="bool" value="false"/>
        <param name="enable_cone_detection" type="bool" value="false"/>
    </node>
    
    <!-- Visualization - RViz-->
    <!-- node name="rviz" pkg="rviz" type="rviz" args="-d $(find orb_slam3_ros)/config/orb_slam3_no_imu.rviz" output="screen" /-->

    <!--node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_orb_slam3" output="screen" ns="orb_slam3_ros" >
        <param name="/target_frame_name" value="/map" />
        <param name="/source_frame_name" value="/camera" />
        <param name="/trajectory_update_rate" value="20.0" />
        <param name="/trajectory_publish_rate" value="20.0" />
    </node-->
</launch>
